src/Struct/FindologicProduct.php

```diff
@@ -350,70 +498,149 @@ public function getProperties(): array
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('purchaseunit');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
 
-    /**
-     * @throws ProductHasNoPricesException
-     */
-    protected function setPrices(): void
-    {
-        $this->setVariantPrices();
-        $this->setProductPrices();
-    }
+        if ($this->product->getReferenceUnit()) {
+            $value = (string)$this->product->getReferenceUnit();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('referenceunit');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
 
-    protected function setDescription(): void
-    {
-        if ($this->product->getTranslation('description')) {
-            $this->description = Utils::cleanString($this->product->getTranslation('description'));
+        if ($this->product->getPackUnit()) {
+            $value = (string)$this->product->getPackUnit();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('packunit');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
+
+        if ($this->product->getStock()) {
+            $value = (string)$this->product->getStock();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('stock');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
+
+        if ($this->product->getAvailableStock()) {
+            $value = (string)$this->product->getAvailableStock();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('availableStock');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
+
+        if ($this->product->getWeight()) {
+            $value = (string)$this->product->getWeight();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('weight');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
+
+        if ($this->product->getWidth()) {
+            $value = (string)$this->product->getWidth();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('width');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
+
+        if ($this->product->getHeight()) {
+            $value = (string)$this->product->getHeight();
+            if (!Utils::isEmpty($value)) {
+                $property = new Property('height');
+                $property->addValue($value);
+                $this->properties[] = $property;
+            }
+        }
```

Comment:
These blocks appear quite often. You probably should add a method that adds the value to `$this->properties` if it's not empty.